akka-actor/src/main/scala/akka/actor/Scheduler.scala

```diff
@@ -42,14 +47,244 @@ private final case class SchedulerException(msg: String) extends akka.AkkaExcept
    * frequency. E.g. if you would like a message to be sent immediately and
-   * thereafter every 500ms you would set delay=Duration.Zero and
-   * interval=Duration(500, TimeUnit.MILLISECONDS)
+   * thereafter every 500ms you would set `delay=Duration.Zero` and
+   * `interval=Duration(500, TimeUnit.MILLISECONDS)`
    *
-   * Java & Scala API
+   * Note: For scheduling within actors `with Timers` should be preferred.
    */
   @silent
+  final def scheduleWithFixedDelay(
+      initialDelay: FiniteDuration,
+      delay: FiniteDuration,
+      receiver: ActorRef,
+      message: Any)(
+      implicit
+      executor: ExecutionContext,
+      sender: ActorRef = Actor.noSender): Cancellable = {
+    scheduleWithFixedDelay(initialDelay, delay, new Runnable {
+      def run(): Unit = {
+        receiver ! message
+        if (receiver.isTerminated)
+          throw SchedulerException("timer active for terminated actor")
+      }
+    })
+  }
+
+  /**
+   * FIXME docs
+   * Java API: Schedules a message to be sent repeatedly with an initial delay and
+   * frequency. E.g. if you would like a message to be sent immediately and
+   * thereafter every 500ms you would set `delay=Duration.ZERO` and
+   * `interval=Duration.ofMillis(500)`
+   *
+   * Note: For scheduling within actors `AbstractActorWithTimers` should be preferred.
+   */
+  final def scheduleWithFixedDelay(
+      initialDelay: java.time.Duration,
+      delay: java.time.Duration,
+      receiver: ActorRef,
+      message: Any,
+      executor: ExecutionContext,
+      sender: ActorRef): Cancellable = {
+    import JavaDurationConverters._
+    scheduleWithFixedDelay(initialDelay.asScala, delay.asScala, receiver, message)(executor, sender)
+  }
+
+  /**
+   * Scala API: Schedules a `Runnable` to be run repeatedly with an initial delay and
+   * a frequency. E.g. if you would like the function to be run after 2
+   * seconds and thereafter every 100ms you would set `delay=Duration(2, TimeUnit.SECONDS)`
+   * and `interval=Duration(100, TimeUnit.MILLISECONDS)`. If
+   * the execution of the runnable takes longer than the interval, the
+   * subsequent execution will start immediately after the prior one completes
+   * (there will be no overlap of executions of the runnable). In such cases,
+   * the actual execution interval will differ from the interval passed to this
+   * method.
+   *
+   * If the `Runnable` throws an exception the repeated scheduling is aborted,
+   * i.e. the function will not be invoked any more.
+   *
+   * @throws IllegalArgumentException if the given delays exceed the maximum
+   * reach (calculated as: `delay / tickNanos > Int.MaxValue`).
+   *
+   * Note: For scheduling within actors `with Timers` should be preferred.
+   */
+  def scheduleAtFixedRate(initialDelay: FiniteDuration, interval: FiniteDuration, runnable: Runnable)(
+      implicit executor: ExecutionContext): Cancellable =
+    schedule(initialDelay, interval, runnable)(executor)
```

Comment:
Delegating to the old method here for compatibility but it will be somewhat strange since the old method should be deprecated. Then custom implementations have to either continue implementing the deprecated `schedule` method or they have to implement both old `schedule` and this `scheduleAtFixedRate`.

This should have been abstract if we hadn't have to consider compatibility.

hmm... ?

Reply:
Second thought. This doesn't have to abstract. We should be able to implement it here in terms of `scheduleOnce` in the same way as it's currently implemented in `LARS.schedule`.

I'll make it final.

Reply:
What was the solution here? Will custom schedulers still work or do they have to be recompiled / changed?

Reply:
Ah, I see all new methods are implemented in terms of others.